lib/model.js

```diff
@@ -3463,10 +3463,18 @@ Model.$__insertMany = function(arr, options, callback) {
             return callback(err);
           }
 
-          callback(null, docs);
+          if (lean) {
```

Comment:
The problem is that in queries, `lean()` avoids hydrating documents. As opposed to here in `insertMany()`, as implemented, we're hydrating the document and then converting it back to a [POJO](https://masteringjs.io/tutorials/fundamentals/pojo). So there's no performance benefit here.

I'd say the better approach would be to avoid calling `doc = new _this(doc);`. It might be better to use `Model.validate()` to validate docs if `lean` is set. Or perhaps just skip validation entirely with `insertMany()` with `lean`.